### PR TITLE
Ticket 3606: Added remote JMX access to the GUI

### PR DESCRIPTION
**Pull request by DominicOram**
### Description of work

Added the JMX port access to the GUI, allowing us to remotely monitor memory and CPU use better.

@FreddieAkeroyd and I looked into adding firewall security on this for the instruments. Unfortunately it dynamically allocates ports as it runs so cannot be easily firewalled. It was decided that the risk in general is low but could be revisited at a later date.

### Ticket

https://github.com/ISISComputingGroup/IBEX/issues/3606

### Acceptance criteria

See documentation at https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Java-Resource-Usage-Tools and https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Debugging-memory-leaks-in-the-IBEX-GUI. 

Confirm that:
* You can remotely access the client with this PR 
* It causes no larger load than usual when externally connected
* You can run multiple clients at the same time

### Unit tests

N/A

### System tests

N/A

### Documentation
Links above

---

#### Code Review

- [ ] Is the code of an acceptable quality?
- [ ] Do the changes function as described and is it robust?
- [ ] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev)?

### Final Steps
- [ ] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

### CODE REVIEW

**Comment by aaron-long** [comment]
- [ ] Update the Wiki to include some of the dangers resulting from GC or profiling on remote instruments.
- [ ] The eclipse client is failing to build due to an API error, can you fix this.

**Comment by DominicOram** [comment]
Changed do this in a more 'correct' way, it means the url to connect to is now ugly though :(

**Comment by aaron-long** [comment]
> 
> 
> Changed do this in a more 'correct' way, it means the url to connect to is now ugly though :(

Yea, it's kind of gross. But It's developer facing and is a working solution that we can maybe clean up later (I've put in a ticket).

File: base/uk.ac.stfc.isis.ibex.e4.product/src/uk/ac/stfc/isis/ibex/e4/product/JMXServer.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	/**
+	 * Starts the server.
+	 */
+	public static void startJMXServer() {
+		try {
+			MBeanServer mbs = ManagementFactory.getPlatformMBeanServer();
+			JMXServiceURL url = new JMXServiceURL("rmi", null, 0);
+			JMXConnectorServer cs = JMXConnectorServerFactory.newJMXConnectorServer(url, null, mbs);
+			cs.start();
+		    System.out.println("JMX url: " + cs.getAddress().toString());
```
Comment (Tom-Willemsen) [inline base/uk.ac.stfc.isis.ibex.e4.product/src/uk/ac/stfc/isis/ibex/e4/product/JMXServer.java]: Can we use a logger rather than `println`?

**Pull request merged**